The job **`dist-x86_64-msvc`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/5178350764/jobs/9329684343) [(plain)](https://github.com/rust-lang-ci/rust/commit/95eed180b590461cd9fde63fae8e156748bc9f5e/checks/14011295980/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
Updating files:  99% (41115/41523)
Updating files:  99% (41342/41523)
Updating files: 100% (41523/41523)
Updating files: 100% (41523/41523), done.
branch 'try' set up to track 'origin/try'.
Switched to a new branch 'try'
[command]"C:\Program Files\Git\bin\git.exe" log -1 --format='%H'
'95eed180b590461cd9fde63fae8e156748bc9f5e'
##[group]Run src/ci/scripts/setup-environment.sh
src/ci/scripts/setup-environment.sh
---
file:.git/config remote.origin.url=https://github.com/rust-lang-ci/rust
file:.git/config remote.origin.fetch=+refs/heads/*:refs/remotes/origin/*
file:.git/config gc.auto=0
file:.git/config http.https://github.com/.extraheader=AUTHORIZATION: basic ***
file:.git/config branch.try.remote=origin
file:.git/config branch.try.merge=refs/heads/try
file:.git/config submodule.library/backtrace.url=https://github.com/rust-lang/backtrace-rs.git
file:.git/config submodule.library/stdarch.active=true
file:.git/config submodule.library/stdarch.url=https://github.com/rust-lang/stdarch.git
file:.git/config submodule.src/doc/book.active=true
---
   Compiling clap_complete v4.2.2
   Compiling xz2 v0.1.6
    Finished dev [unoptimized] target(s) in 35.55s
##[endgroup]
warning: some crates are on edition 2021 which defaults to `resolver = "2"`, but virtual workspaces default to `resolver = "1"`
note: to keep the current resolver, specify `workspace.resolver = "1"` in the workspace root's manifest
note: to use the edition 2021 resolver, specify `workspace.resolver = "2"` in the workspace root's manifest
#=#=#                                                                         
##O#-#                                                                        

                                                                           0.0%
---
failures:

---- [codegen] tests\codegen\vec-shrink-panik.rs#old stdout ----

error in revision `old`: verification with 'FileCheck' failed
status: exit code: 1
command: PATH=";C:\Program Files (x86)\Windows Kits\10\bin\x64;C:\Program Files (x86)\Windows Kits\10\bin\10.0.22621.0\x64;C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.29.30133\bin\HostX64\x64;C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.29.30133\bin\HostX64\x64;C:\a\rust\rust\build\x86_64-pc-windows-msvc\stage0-bootstrap-tools\x86_64-pc-windows-msvc\release\deps;C:\a\rust\rust\build\optimized-dist\rustc-nightly-x86_64-pc-windows-msvc\rustc\bin;C:\a\rust\rust\ninja;C:\a\rust\rust\msys2\mingw64\bin;C:\hostedtoolcache\windows\Python\3.11.3\x64\Scripts;C:\hostedtoolcache\windows\Python\3.11.3\x64;C:\msys64\usr\bin;C:\a\rust\rust\sccache;C:\Program Files\MongoDB\Server\5.0\bin;C:\aliyun-cli;C:\vcpkg;C:\cf-cli;C:\Program Files (x86)\NSIS;C:\tools\zstd;C:\Program Files\Mercurial;C:\hostedtoolcache\windows\stack\2.9.3\x64;C:\cabal\bin;C:\ghcup\bin;C:\Program Files\dotnet;C:\mysql\bin;C:\Program Files\R\R-4.3.0\bin\x64;C:\SeleniumWebDrivers\GeckoDriver;C:\Program Files (x86)\sbt\bin;C:\Program Files (x86)\GitHub CLI;C:\Program Files\Git\bin;C:\Program Files (x86)\pipx_bin;C:\npm\prefix;C:\hostedtoolcache\windows\go\1.20.4\x64\bin;C:\hostedtoolcache\windows\Python\3.7.9\x64\Scripts;C:\hostedtoolcache\windows\Python\3.7.9\x64;C:\hostedtoolcache\windows\Ruby\2.5.9\x64\bin;C:\Program Files\OpenSSL\bin;C:\tools\kotlinc\bin;C:\hostedtoolcache\windows\Java_Temurin-Hotspot_jdk\8.0.372-7\x64\bin;C:\Program Files\ImageMagick-7.1.1-Q16-HDRI;C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\wbin;C:\ProgramData\kind;C:\Program Files\Eclipse Foundation\jdk-8.0.302.8-hotspot\bin;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0;C:\Windows\System32\OpenSSH;C:\ProgramData\Chocolatey\bin;C:\Program Files\PowerShell\7;C:\Program Files\Microsoft\Web Platform Installer;C:\Program Files\Microsoft SQL Server\130\Tools\Binn;C:\Program Files\Microsoft SQL Server\Client SDK\ODBC\170\Tools\Binn;C:\Program Files (x86)\Windows Kits\10\Windows Performance Toolkit;C:\Program Files (x86)\Microsoft SQL Server\110\DTS\Binn;C:\Program Files (x86)\Microsoft SQL Server\120\DTS\Binn;C:\Program Files (x86)\Microsoft SQL Server\130\DTS\Binn;C:\Program Files (x86)\Microsoft SQL Server\140\DTS\Binn;C:\Program Files (x86)\Microsoft SQL Server\150\DTS\Binn;C:\Program Files (x86)\Microsoft SQL Server\160\DTS\Binn;C:\Strawberry\c\bin;C:\Strawberry\perl\site\bin;C:\Strawberry\perl\bin;C:\ProgramData\chocolatey\lib\pulumi\tools\Pulumi\bin;C:\Program Files\TortoiseSVN\bin;C:\Program Files\CMake\bin;C:\ProgramData\chocolatey\lib\maven\apache-maven-3.8.7\bin;C:\Program Files\Microsoft Service Fabric\bin\Fabric\Fabric.Code;C:\Program Files\Microsoft SDKs\Service Fabric\Tools\ServiceFabricLocalClusterManager;C:\Program Files\nodejs;C:\Program Files\Git\cmd;C:\Program Files\Git\mingw64\bin;C:\Program Files\Git\usr\bin;C:\Program Files\GitHub CLI;C:\tools\php;C:\Program Files (x86)\sbt\bin;C:\SeleniumWebDrivers\ChromeDriver;C:\SeleniumWebDrivers\EdgeDriver;C:\Program Files\Amazon\AWSCLIV2;C:\Program Files\Amazon\SessionManagerPlugin\bin;C:\Program Files\Amazon\AWSSAMCLI\bin;C:\Program Files (x86)\Google\Cloud SDK\google-cloud-sdk\bin;C:\Program Files (x86)\Microsoft BizTalk Server;C:\Program Files\LLVM\bin;C:\Users\runneradmin\.dotnet\tools;C:\Users\runneradmin\.cargo\bin;C:\Users\runneradmin\AppData\Local\Microsoft\WindowsApps" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\ci-llvm\\bin\\FileCheck.exe" "--input-file" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\test\\codegen\\vec-shrink-panik.old\\vec-shrink-panik.ll" "C:\\a\\rust\\rust\\tests\\codegen\\vec-shrink-panik.rs" "--allow-unused-prefixes" "--check-prefixes" "CHECK,MSVC,old" "--dump-input-context" "100"
stdout: none
C:\a\rust\rust\tests\codegen\vec-shrink-panik.rs:29:10: error: old: expected string not found in input
 // old: filter
         ^
         ^
C:\a\rust\rust\build\x86_64-pc-windows-msvc\test\codegen\vec-shrink-panik.old\vec-shrink-panik.ll:99:32: note: scanning from here
define { ptr, i64 } @issue71861(ptr noalias nocapture noundef dereferenceable(24) %vec) unnamed_addr #0 personality ptr @__CxxFrameHandler3 {
                               ^
C:\a\rust\rust\build\x86_64-pc-windows-msvc\test\codegen\vec-shrink-panik.old\vec-shrink-panik.ll:151:42: note: possible intended match here
 cleanupret from %cleanuppad.i unwind to caller
C:\a\rust\rust\tests\codegen\vec-shrink-panik.rs:44:10: error: old: expected string not found in input
 // old: filter
         ^
         ^
C:\a\rust\rust\build\x86_64-pc-windows-msvc\test\codegen\vec-shrink-panik.old\vec-shrink-panik.ll:161:32: note: scanning from here
define { ptr, i64 } @issue75636(ptr noalias noundef nonnull readonly align 8 %iter.0, i64 noundef %iter.1) unnamed_addr #0 personality ptr @__CxxFrameHandler3 {
                               ^
C:\a\rust\rust\build\x86_64-pc-windows-msvc\test\codegen\vec-shrink-panik.old\vec-shrink-panik.ll:161:78: note: possible intended match here
define { ptr, i64 } @issue75636(ptr noalias noundef nonnull readonly align 8 %iter.0, i64 noundef %iter.1) unnamed_addr #0 personality ptr @__CxxFrameHandler3 {


Input file: C:\a\rust\rust\build\x86_64-pc-windows-msvc\test\codegen\vec-shrink-panik.old\vec-shrink-panik.ll
Check file: C:\a\rust\rust\tests\codegen\vec-shrink-panik.rs

-dump-input=help explains the following input dump.
Input was:
<<<<<<
<<<<<<
            1: ; ModuleID = 'vec_shrink_panik.e8006217fa00f06d-cgu.0' 
            2: source_filename = "vec_shrink_panik.e8006217fa00f06d-cgu.0" 
            3: target datalayout = "e-m:w-p270:32:32-p271:32:32-p272:64:64-i64:64-f80:128-n8:16:32:64-S128" 
            4: target triple = "x86_64-pc-windows-msvc" 
            5:  
            6: %"alloc::vec::Vec<u32>" = type { { ptr, i64 }, i64 } 
            7: %"alloc::vec::Vec<&str>" = type { { ptr, i64 }, i64 } 
            8:  
            9: @__rust_no_alloc_shim_is_unstable = external global i8 
           11: ; core::ptr::drop_in_place<alloc::vec::Vec<u32>> 
           11: ; core::ptr::drop_in_place<alloc::vec::Vec<u32>> 
           12: ; Function Attrs: uwtable 
           13: define internal fastcc void @"_ZN4core3ptr47drop_in_place$LT$alloc..vec..Vec$LT$u32$GT$$GT$17h3e6352b4d312dad2E"(ptr noalias nocapture noundef readonly align 8 dereferenceable(24) %_1) unnamed_addr #0 personality ptr @__CxxFrameHandler3 { 
           14: bb4: 
           15:  tail call void @llvm.experimental.noalias.scope.decl(metadata !1) 
           16:  tail call void @llvm.experimental.noalias.scope.decl(metadata !4) 
           17:  %0 = getelementptr inbounds { ptr, i64 }, ptr %_1, i64 0, i32 1 
           18:  %_4.i.i.i = load i64, ptr %0, align 8, !alias.scope !7, !noalias !10, !noundef !12 
           19:  %_3.i.i.i = icmp eq i64 %_4.i.i.i, 0 
           20:  br i1 %_3.i.i.i, label %"_ZN4core3ptr54drop_in_place$LT$alloc..raw_vec..RawVec$LT$u32$GT$$GT$17h1c51bca9c0b39455E.exit", label %bb1.i.i.i 
           21:  
           22: bb1.i.i.i: ; preds = %bb4 
           23:  %1 = shl nuw i64 %_4.i.i.i, 2 
           24:  %self3.i.i.i = load ptr, ptr %_1, align 8, !alias.scope !7, !noalias !10, !nonnull !12, !noundef !12 
           25:  tail call void @__rust_dealloc(ptr noundef nonnull %self3.i.i.i, i64 noundef %1, i64 noundef 4) #10, !noalias !13 
           26:  br label %"_ZN4core3ptr54drop_in_place$LT$alloc..raw_vec..RawVec$LT$u32$GT$$GT$17h1c51bca9c0b39455E.exit" 
           27:  
           28: "_ZN4core3ptr54drop_in_place$LT$alloc..raw_vec..RawVec$LT$u32$GT$$GT$17h1c51bca9c0b39455E.exit": ; preds = %bb4, %bb1.i.i.i 
           29:  ret void 
           30: } 
           31:  
           32: ; core::ptr::drop_in_place<alloc::vec::Vec<&str>> 
           33: ; Function Attrs: uwtable 
           34: define internal fastcc void @"_ZN4core3ptr51drop_in_place$LT$alloc..vec..Vec$LT$$RF$str$GT$$GT$17he6ad928979051c70E"(ptr noalias nocapture noundef readonly align 8 dereferenceable(24) %_1) unnamed_addr #0 personality ptr @__CxxFrameHandler3 { 
           35: bb4: 
           36:  tail call void @llvm.experimental.noalias.scope.decl(metadata !14) 
           37:  tail call void @llvm.experimental.noalias.scope.decl(metadata !17) 
           38:  %0 = getelementptr inbounds { ptr, i64 }, ptr %_1, i64 0, i32 1 
           39:  %_4.i.i.i = load i64, ptr %0, align 8, !alias.scope !20, !noalias !23, !noundef !12 
           40:  %_3.i.i.i = icmp eq i64 %_4.i.i.i, 0 
           41:  br i1 %_3.i.i.i, label %"_ZN4core3ptr58drop_in_place$LT$alloc..raw_vec..RawVec$LT$$RF$str$GT$$GT$17h8c063336958f0a51E.exit", label %bb1.i.i.i 
           42:  
           43: bb1.i.i.i: ; preds = %bb4 
           44:  %1 = shl nuw i64 %_4.i.i.i, 4 
           45:  %self3.i.i.i = load ptr, ptr %_1, align 8, !alias.scope !20, !noalias !23, !nonnull !12, !noundef !12 
           46:  tail call void @__rust_dealloc(ptr noundef nonnull %self3.i.i.i, i64 noundef %1, i64 noundef 8) #10, !noalias !25 
           47:  br label %"_ZN4core3ptr58drop_in_place$LT$alloc..raw_vec..RawVec$LT$$RF$str$GT$$GT$17h8c063336958f0a51E.exit" 
           48:  
           49: "_ZN4core3ptr58drop_in_place$LT$alloc..raw_vec..RawVec$LT$$RF$str$GT$$GT$17h8c063336958f0a51E.exit": ; preds = %bb4, %bb1.i.i.i 
           50:  ret void 
           51: } 
           52:  
           53: ; Function Attrs: uwtable 
           54: define void @shrink_to_fit(ptr noalias nocapture noundef align 8 dereferenceable(24) %vec) unnamed_addr #0 { 
           55: start: 
           56:  tail call void @llvm.experimental.noalias.scope.decl(metadata !26) 
           57:  %0 = getelementptr inbounds { ptr, i64 }, ptr %vec, i64 0, i32 1 
           58:  %1 = load i64, ptr %0, align 8, !alias.scope !26, !noundef !12 
           59:  %2 = getelementptr inbounds %"alloc::vec::Vec<u32>", ptr %vec, i64 0, i32 1 
           60:  %_5.i = load i64, ptr %2, align 8, !alias.scope !26, !noundef !12 
           61:  %_2.i = icmp ugt i64 %1, %_5.i 
           62:  br i1 %_2.i, label %bb4.i.i, label %"_ZN5alloc3vec16Vec$LT$T$C$A$GT$13shrink_to_fit17h59e5e1579f7e9fd6E.exit" 
           63:  
           64: bb4.i.i: ; preds = %start 
           65:  tail call void @llvm.experimental.noalias.scope.decl(metadata !29) 
           66:  %3 = shl nuw i64 %1, 2 
           67:  %self3.i.i.i = load ptr, ptr %vec, align 8, !alias.scope !32, !noalias !35, !nonnull !12, !noundef !12 
           68:  %4 = icmp eq i64 %_5.i, 0 
           69:  br i1 %4, label %bb1.i.i.i.i, label %"_ZN63_$LT$alloc..alloc..Global$u20$as$u20$core..alloc..Allocator$GT$6shrink17h07a4bd2cb2e9e15aE.exit.i.i" 
           70:  
           71: bb1.i.i.i.i: ; preds = %bb4.i.i 
           72:  tail call void @__rust_dealloc(ptr noundef nonnull %self3.i.i.i, i64 noundef %3, i64 noundef 4) #10, !noalias !37 
           73:  br label %_ZN5alloc7raw_vec14handle_reserve17hb9da0870c25f94d3E.exit.i 
           74:  
           75: "_ZN63_$LT$alloc..alloc..Global$u20$as$u20$core..alloc..Allocator$GT$6shrink17h07a4bd2cb2e9e15aE.exit.i.i": ; preds = %bb4.i.i 
           76:  %5 = shl nuw i64 %_5.i, 2 
           77:  %_22.i.i.i = icmp ule i64 %5, %3 
           78:  tail call void @llvm.assume(i1 %_22.i.i.i) 
           79:  %raw_ptr.i.i.i = tail call noundef align 4 ptr @__rust_realloc(ptr noundef nonnull %self3.i.i.i, i64 noundef %3, i64 noundef 4, i64 noundef %5) #10, !noalias !37 
           80:  %6 = icmp eq ptr %raw_ptr.i.i.i, null 
           81:  br i1 %6, label %bb6.i.i, label %_ZN5alloc7raw_vec14handle_reserve17hb9da0870c25f94d3E.exit.i 
           82:  
           83: bb6.i.i: ; preds = %"_ZN63_$LT$alloc..alloc..Global$u20$as$u20$core..alloc..Allocator$GT$6shrink17h07a4bd2cb2e9e15aE.exit.i.i" 
           84: ; call alloc::alloc::handle_alloc_error 
           85:  tail call void @_ZN5alloc5alloc18handle_alloc_error17h1ce469003af61c69E(i64 noundef 4, i64 noundef %5) #11, !noalias !26 
           86:  unreachable 
           87:  
           88: _ZN5alloc7raw_vec14handle_reserve17hb9da0870c25f94d3E.exit.i: ; preds = %"_ZN63_$LT$alloc..alloc..Global$u20$as$u20$core..alloc..Allocator$GT$6shrink17h07a4bd2cb2e9e15aE.exit.i.i", %bb1.i.i.i.i 
           89:  %.sroa.0.0.i22.i.i = phi ptr [ inttoptr (i64 4 to ptr), %bb1.i.i.i.i ], [ %raw_ptr.i.i.i, %"_ZN63_$LT$alloc..alloc..Global$u20$as$u20$core..alloc..Allocator$GT$6shrink17h07a4bd2cb2e9e15aE.exit.i.i" ] 
           90:  store ptr %.sroa.0.0.i22.i.i, ptr %vec, align 8, !alias.scope !38 
           91:  store i64 %_5.i, ptr %0, align 8, !alias.scope !38 
           92:  br label %"_ZN5alloc3vec16Vec$LT$T$C$A$GT$13shrink_to_fit17h59e5e1579f7e9fd6E.exit" 
           93:  
           94: "_ZN5alloc3vec16Vec$LT$T$C$A$GT$13shrink_to_fit17h59e5e1579f7e9fd6E.exit": ; preds = %start, %_ZN5alloc7raw_vec14handle_reserve17hb9da0870c25f94d3E.exit.i 
           95:  ret void 
           96: } 
           97:  
           98: ; Function Attrs: uwtable 
           99: define { ptr, i64 } @issue71861(ptr noalias nocapture noundef dereferenceable(24) %vec) unnamed_addr #0 personality ptr @__CxxFrameHandler3 { 
check:29'0                                    X~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ error: no match found
          100: start: 
check:29'0     ~~~~~~~
          101:  tail call void @llvm.experimental.noalias.scope.decl(metadata !41) 
check:29'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          102:  tail call void @llvm.experimental.noalias.scope.decl(metadata !44) 
check:29'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          103:  %0 = getelementptr inbounds { ptr, i64 }, ptr %vec, i64 0, i32 1 
check:29'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          104:  %1 = load i64, ptr %0, align 8, !alias.scope !47, !noundef !12 
check:29'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          105:  %2 = getelementptr inbounds %"alloc::vec::Vec<u32>", ptr %vec, i64 0, i32 1 
check:29'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          106:  %_5.i.i = load i64, ptr %2, align 8, !alias.scope !47, !noundef !12 
Build completed unsuccessfully in 0:02:11
check:29'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          107:  %_2.i.i = icmp ugt i64 %1, %_5.i.i 
check:29'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          108:  br i1 %_2.i.i, label %bb4.i.i.i, label %start.bb2_crit_edge.i 
check:29'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          109:  
check:29'0     ~
          110: start.bb2_crit_edge.i: ; preds = %start 
check:29'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          111:  %value.sroa.0.0.copyload.pre.i = load ptr, ptr %vec, align 8, !alias.scope !41 
check:29'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          112:  br label %"_ZN5alloc3vec16Vec$LT$T$C$A$GT$16into_boxed_slice17hab286dc5286d9e5eE.exit" 
check:29'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          113:  
check:29'0     ~
          114: bb4.i.i.i: ; preds = %start 
check:29'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          115:  tail call void @llvm.experimental.noalias.scope.decl(metadata !48) 
check:29'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          116:  %3 = shl nuw i64 %1, 2 
check:29'0     ~~~~~~~~~~~~~~~~~~~~~~~~
          117:  %self3.i.i.i.i = load ptr, ptr %vec, align 8, !alias.scope !51, !noalias !54, !nonnull !12, !noundef !12 
check:29'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          118:  %4 = icmp eq i64 %_5.i.i, 0 
check:29'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          119:  br i1 %4, label %bb1.i.i.i.i.i, label %"_ZN63_$LT$alloc..alloc..Global$u20$as$u20$core..alloc..Allocator$GT$6shrink17h07a4bd2cb2e9e15aE.exit.i.i.i" 
check:29'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          120:  
check:29'0     ~
          121: bb1.i.i.i.i.i: ; preds = %bb4.i.i.i 
check:29'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          122:  tail call void @__rust_dealloc(ptr noundef nonnull %self3.i.i.i.i, i64 noundef %3, i64 noundef 4) #10, !noalias !56 
check:29'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          123:  br label %_ZN5alloc7raw_vec14handle_reserve17hb9da0870c25f94d3E.exit.i.i 
check:29'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          124:  
check:29'0     ~
          125: "_ZN63_$LT$alloc..alloc..Global$u20$as$u20$core..alloc..Allocator$GT$6shrink17h07a4bd2cb2e9e15aE.exit.i.i.i": ; preds = %bb4.i.i.i 
check:29'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          126:  %5 = shl nuw i64 %_5.i.i, 2 
check:29'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          127:  %_22.i.i.i.i = icmp ule i64 %5, %3 
check:29'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          128:  tail call void @llvm.assume(i1 %_22.i.i.i.i) 
check:29'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          129:  %raw_ptr.i.i.i.i = tail call noundef align 4 ptr @__rust_realloc(ptr noundef nonnull %self3.i.i.i.i, i64 noundef %3, i64 noundef 4, i64 noundef %5) #10, !noalias !56 
check:29'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          130:  %6 = icmp eq ptr %raw_ptr.i.i.i.i, null 
check:29'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          131:  br i1 %6, label %bb6.i.i.i, label %_ZN5alloc7raw_vec14handle_reserve17hb9da0870c25f94d3E.exit.i.i 
check:29'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          132:  
check:29'0     ~
          133: bb6.i.i.i: ; preds = %"_ZN63_$LT$alloc..alloc..Global$u20$as$u20$core..alloc..Allocator$GT$6shrink17h07a4bd2cb2e9e15aE.exit.i.i.i" 
check:29'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          134: ; invoke alloc::alloc::handle_alloc_error 
check:29'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          135:  invoke void @_ZN5alloc5alloc18handle_alloc_error17h1ce469003af61c69E(i64 noundef 4, i64 noundef %5) #11 
check:29'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          136:  to label %.noexc.i unwind label %bb3.i, !noalias !41 
check:29'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          137:  
check:29'0     ~
          138: .noexc.i: ; preds = %bb6.i.i.i 
check:29'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          139:  unreachable 
check:29'0     ~~~~~~~~~~~~~
          140:  
check:29'0     ~
          141: _ZN5alloc7raw_vec14handle_reserve17hb9da0870c25f94d3E.exit.i.i: ; preds = %"_ZN63_$LT$alloc..alloc..Global$u20$as$u20$core..alloc..Allocator$GT$6shrink17h07a4bd2cb2e9e15aE.exit.i.i.i", %bb1.i.i.i.i.i 
check:29'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          142:  %.sroa.0.0.i22.i.i.i = phi ptr [ inttoptr (i64 4 to ptr), %bb1.i.i.i.i.i ], [ %raw_ptr.i.i.i.i, %"_ZN63_$LT$alloc..alloc..Global$u20$as$u20$core..alloc..Allocator$GT$6shrink17h07a4bd2cb2e9e15aE.exit.i.i.i" ] 
check:29'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          143:  store ptr %.sroa.0.0.i22.i.i.i, ptr %vec, align 8, !alias.scope !57 
check:29'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          144:  store i64 %_5.i.i, ptr %0, align 8, !alias.scope !57 
check:29'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          145:  br label %"_ZN5alloc3vec16Vec$LT$T$C$A$GT$16into_boxed_slice17hab286dc5286d9e5eE.exit" 
check:29'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          146:  
check:29'0     ~
          147: bb3.i: ; preds = %bb6.i.i.i 
check:29'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          148:  %cleanuppad.i = cleanuppad within none [] 
check:29'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          149: ; call core::ptr::drop_in_place<alloc::vec::Vec<u32>> 
check:29'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          150:  call fastcc void @"_ZN4core3ptr47drop_in_place$LT$alloc..vec..Vec$LT$u32$GT$$GT$17h3e6352b4d312dad2E"(ptr noalias noundef nonnull align 8 dereferenceable(24) %vec) #12 [ "funclet"(token %cleanuppad.i) ] 
check:29'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          151:  cleanupret from %cleanuppad.i unwind to caller 
check:29'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:29'1                                              ?       possible intended match
          152:  
check:29'0     ~
          153: "_ZN5alloc3vec16Vec$LT$T$C$A$GT$16into_boxed_slice17hab286dc5286d9e5eE.exit": ; preds = %start.bb2_crit_edge.i, %_ZN5alloc7raw_vec14handle_reserve17hb9da0870c25f94d3E.exit.i.i 
check:29'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          154:  %value.sroa.0.0.copyload.i = phi ptr [ %value.sroa.0.0.copyload.pre.i, %start.bb2_crit_edge.i ], [ %.sroa.0.0.i22.i.i.i, %_ZN5alloc7raw_vec14handle_reserve17hb9da0870c25f94d3E.exit.i.i ] 
check:29'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          155:  %7 = insertvalue { ptr, i64 } poison, ptr %value.sroa.0.0.copyload.i, 0 
check:29'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          156:  %8 = insertvalue { ptr, i64 } %7, i64 %_5.i.i, 1 
check:29'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          157:  ret { ptr, i64 } %8 
check:29'0     ~~~~~~~~~~~~~~~~~~~~~
          158: } 
check:29'0     ~~
          159:  
check:29'0     ~
          160: ; Function Attrs: uwtable 
check:29'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~
          161: define { ptr, i64 } @issue75636(ptr noalias noundef nonnull readonly align 8 %iter.0, i64 noundef %iter.1) unnamed_addr #0 personality ptr @__CxxFrameHandler3 { 
check:29'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:44'0                                    X~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ error: no match found
check:44'1                                                                                  ?                                                                                    possible intended match
          162: start: 
check:44'0     ~~~~~~~
          163:  %_2.i = alloca %"alloc::vec::Vec<&str>", align 8 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          164:  %end = getelementptr inbounds { ptr, i64 }, ptr %iter.0, i64 %iter.1 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          165:  call void @llvm.lifetime.start.p0(i64 24, ptr nonnull %_2.i) 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          166:  tail call void @llvm.experimental.noalias.scope.decl(metadata !60) 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          167:  tail call void @llvm.experimental.noalias.scope.decl(metadata !63) 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          168:  tail call void @llvm.experimental.noalias.scope.decl(metadata !66) 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          169:  tail call void @llvm.experimental.noalias.scope.decl(metadata !69) 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          170:  %end.idx = shl nuw nsw i64 %iter.1, 4 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          171:  %_5.i.i.i.i.i.i = icmp eq i64 %iter.1, 0 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          172:  br i1 %_5.i.i.i.i.i.i, label %"_ZN107_$LT$alloc..boxed..Box$LT$$u5b$I$u5d$$GT$$u20$as$u20$core..iter..traits..collect..FromIterator$LT$I$GT$$GT$9from_iter17h85113226ea47dfa0E.exit", label %bb4.i.i.i.i.i.i 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          173:  
check:44'0     ~
          174: bb4.i.i.i.i.i.i: ; preds = %start 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          175:  %_5.i.i.i.i.i.i.i = icmp sgt i64 %iter.1, -1 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          176:  %0 = lshr i64 %iter.1, 56 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~
          177:  %1 = and i64 %0, 8 
check:44'0     ~~~~~~~~~~~~~~~~~~~~
          178:  %.sroa.0.0.i.i.i.i.i.i.i = xor i64 %1, 8 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          179:  br i1 %_5.i.i.i.i.i.i.i, label %"_ZN63_$LT$alloc..alloc..Global$u20$as$u20$core..alloc..Allocator$GT$8allocate17hbca294f9c89adf79E.exit.i.i.i.i.i.i", label %bb5.i.i.i.i.i.i 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          180:  
check:44'0     ~
          181: bb5.i.i.i.i.i.i: ; preds = %bb4.i.i.i.i.i.i 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          182: ; call alloc::raw_vec::capacity_overflow 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          183:  tail call void @_ZN5alloc7raw_vec17capacity_overflow17h3c7ef808270c8afeE() #11, !noalias !72 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          184:  unreachable 
check:44'0     ~~~~~~~~~~~~~
          185:  
check:44'0     ~
          186: "_ZN63_$LT$alloc..alloc..Global$u20$as$u20$core..alloc..Allocator$GT$8allocate17hbca294f9c89adf79E.exit.i.i.i.i.i.i": ; preds = %bb4.i.i.i.i.i.i 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          187:  %2 = load volatile i8, ptr @__rust_no_alloc_shim_is_unstable, align 1, !noalias !72 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          188:  %3 = tail call noundef ptr @__rust_alloc(i64 noundef %end.idx, i64 noundef %.sroa.0.0.i.i.i.i.i.i.i) #10, !noalias !72 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          189:  %4 = icmp eq ptr %3, null 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~
          190:  br i1 %4, label %bb13.i.i.i.i.i.i, label %bb3.i.i.i.i.i.i.i.i.i.i 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          191:  
check:44'0     ~
          192: bb13.i.i.i.i.i.i: ; preds = %"_ZN63_$LT$alloc..alloc..Global$u20$as$u20$core..alloc..Allocator$GT$8allocate17hbca294f9c89adf79E.exit.i.i.i.i.i.i" 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          193: ; call alloc::alloc::handle_alloc_error 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          194:  tail call void @_ZN5alloc5alloc18handle_alloc_error17h1ce469003af61c69E(i64 noundef %.sroa.0.0.i.i.i.i.i.i.i, i64 noundef %end.idx) #11, !noalias !72 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          195:  unreachable 
check:44'0     ~~~~~~~~~~~~~
          196:  
check:44'0     ~
          197: bb3.i.i.i.i.i.i.i.i.i.i: ; preds = %"_ZN63_$LT$alloc..alloc..Global$u20$as$u20$core..alloc..Allocator$GT$8allocate17hbca294f9c89adf79E.exit.i.i.i.i.i.i", %bb3.i.i.i.i.i.i.i.i.i.i 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          198:  %5 = phi i64 [ %8, %bb3.i.i.i.i.i.i.i.i.i.i ], [ 0, %"_ZN63_$LT$alloc..alloc..Global$u20$as$u20$core..alloc..Allocator$GT$8allocate17hbca294f9c89adf79E.exit.i.i.i.i.i.i" ] 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          199:  %self.sroa.0.05.i.i.i.i.i.i.i.i.i.i = phi ptr [ %_32.i.i.i.i.i.i.i.i.i.i.i, %bb3.i.i.i.i.i.i.i.i.i.i ], [ %iter.0, %"_ZN63_$LT$alloc..alloc..Global$u20$as$u20$core..alloc..Allocator$GT$8allocate17hbca294f9c89adf79E.exit.i.i.i.i.i.i" ] 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          200:  %_32.i.i.i.i.i.i.i.i.i.i.i = getelementptr inbounds { ptr, i64 }, ptr %self.sroa.0.05.i.i.i.i.i.i.i.i.i.i, i64 1 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          201:  tail call void @llvm.experimental.noalias.scope.decl(metadata !73) 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          202:  %elt.0.i.i.i.i.i.i.i.i.i.i.i = load ptr, ptr %self.sroa.0.05.i.i.i.i.i.i.i.i.i.i, align 8, !alias.scope !73, !noalias !76, !nonnull !12, !align !88, !noundef !12 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          203:  %6 = getelementptr inbounds { ptr, i64 }, ptr %self.sroa.0.05.i.i.i.i.i.i.i.i.i.i, i64 0, i32 1 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          204:  %elt.1.i.i.i.i.i.i.i.i.i.i.i = load i64, ptr %6, align 8, !alias.scope !73, !noalias !76, !noundef !12 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          205:  %dst.i.i.i.i.i.i.i.i.i.i.i.i.i = getelementptr inbounds { ptr, i64 }, ptr %3, i64 %5 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          206:  store ptr %elt.0.i.i.i.i.i.i.i.i.i.i.i, ptr %dst.i.i.i.i.i.i.i.i.i.i.i.i.i, align 8, !noalias !89 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          207:  %7 = getelementptr inbounds { ptr, i64 }, ptr %3, i64 %5, i32 1 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          208:  store i64 %elt.1.i.i.i.i.i.i.i.i.i.i.i, ptr %7, align 8, !noalias !96 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          209:  %8 = add nuw nsw i64 %5, 1 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          210:  %9 = icmp eq ptr %_32.i.i.i.i.i.i.i.i.i.i.i, %end 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          211:  br i1 %9, label %_ZN4core4iter6traits8iterator8Iterator7collect17ha6b288539e8228cbE.exit.i, label %bb3.i.i.i.i.i.i.i.i.i.i 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          212:  
check:44'0     ~
          213: _ZN4core4iter6traits8iterator8Iterator7collect17ha6b288539e8228cbE.exit.i: ; preds = %bb3.i.i.i.i.i.i.i.i.i.i 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          214:  store ptr %3, ptr %_2.i, align 8, !alias.scope !72 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          215:  %vector.sroa.5.0..sroa_idx.i.i.i.i.i = getelementptr inbounds i8, ptr %_2.i, i64 8 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          216:  store i64 %iter.1, ptr %vector.sroa.5.0..sroa_idx.i.i.i.i.i, align 8, !alias.scope !72 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          217:  %vector.sroa.6.0..sroa_idx.i.i.i.i.i = getelementptr inbounds i8, ptr %_2.i, i64 16 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          218:  store i64 %8, ptr %vector.sroa.6.0..sroa_idx.i.i.i.i.i, align 8, !alias.scope !72 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          219:  %_2.i.i.i = icmp ult i64 %8, %iter.1 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          220:  br i1 %_2.i.i.i, label %"_ZN63_$LT$alloc..alloc..Global$u20$as$u20$core..alloc..Allocator$GT$6shrink17h07a4bd2cb2e9e15aE.exit.i.i.i.i", label %"_ZN107_$LT$alloc..boxed..Box$LT$$u5b$I$u5d$$GT$$u20$as$u20$core..iter..traits..collect..FromIterator$LT$I$GT$$GT$9from_iter17h85113226ea47dfa0E.exit" 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          221:  
check:44'0     ~
          222: "_ZN63_$LT$alloc..alloc..Global$u20$as$u20$core..alloc..Allocator$GT$6shrink17h07a4bd2cb2e9e15aE.exit.i.i.i.i": ; preds = %_ZN4core4iter6traits8iterator8Iterator7collect17ha6b288539e8228cbE.exit.i 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          223:  %10 = shl nuw i64 %8, 4 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~
          224:  %_22.i.i.i.i.i = icmp ule i64 %10, %end.idx 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          225:  tail call void @llvm.assume(i1 %_22.i.i.i.i.i) 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          226:  %raw_ptr.i.i.i.i.i = tail call noundef align 8 ptr @__rust_realloc(ptr noundef nonnull %3, i64 noundef %end.idx, i64 noundef 8, i64 noundef %10) #10, !noalias !97 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          227:  %11 = icmp eq ptr %raw_ptr.i.i.i.i.i, null 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          228:  br i1 %11, label %bb6.i.i.i.i, label %"_ZN107_$LT$alloc..boxed..Box$LT$$u5b$I$u5d$$GT$$u20$as$u20$core..iter..traits..collect..FromIterator$LT$I$GT$$GT$9from_iter17h85113226ea47dfa0E.exit" 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          229:  
check:44'0     ~
          230: bb6.i.i.i.i: ; preds = %"_ZN63_$LT$alloc..alloc..Global$u20$as$u20$core..alloc..Allocator$GT$6shrink17h07a4bd2cb2e9e15aE.exit.i.i.i.i" 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          231: ; invoke alloc::alloc::handle_alloc_error 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          232:  invoke void @_ZN5alloc5alloc18handle_alloc_error17h1ce469003af61c69E(i64 noundef 8, i64 noundef %10) #11 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          233:  to label %.noexc.i.i unwind label %bb3.i.i, !noalias !104 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          234:  
check:44'0     ~
          235: .noexc.i.i: ; preds = %bb6.i.i.i.i 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          236:  unreachable 
check:44'0     ~~~~~~~~~~~~~
          237:  
check:44'0     ~
          238: bb3.i.i: ; preds = %bb6.i.i.i.i 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          239:  %cleanuppad.i.i = cleanuppad within none [] 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          240: ; call core::ptr::drop_in_place<alloc::vec::Vec<&str>> 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          241:  call fastcc void @"_ZN4core3ptr51drop_in_place$LT$alloc..vec..Vec$LT$$RF$str$GT$$GT$17he6ad928979051c70E"(ptr noalias noundef nonnull align 8 dereferenceable(24) %_2.i) #12 [ "funclet"(token %cleanuppad.i.i) ] 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          242:  cleanupret from %cleanuppad.i.i unwind to caller 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          243:  
check:44'0     ~
          244: "_ZN107_$LT$alloc..boxed..Box$LT$$u5b$I$u5d$$GT$$u20$as$u20$core..iter..traits..collect..FromIterator$LT$I$GT$$GT$9from_iter17h85113226ea47dfa0E.exit": ; preds = %"_ZN63_$LT$alloc..alloc..Global$u20$as$u20$core..alloc..Allocator$GT$6shrink17h07a4bd2cb2e9e15aE.exit.i.i.i.i", %start, %_ZN4core4iter6traits8iterator8Iterator7collect17ha6b288539e8228cbE.exit.i 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          245:  %_2.i.i.i.i.i.i.i.i.i.i.i.i.i.i8.i = phi i64 [ %8, %_ZN4core4iter6traits8iterator8Iterator7collect17ha6b288539e8228cbE.exit.i ], [ 0, %start ], [ %8, %"_ZN63_$LT$alloc..alloc..Global$u20$as$u20$core..alloc..Allocator$GT$6shrink17h07a4bd2cb2e9e15aE.exit.i.i.i.i" ] 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          246:  %value.sroa.0.0.copyload.i.i = phi ptr [ %3, %_ZN4core4iter6traits8iterator8Iterator7collect17ha6b288539e8228cbE.exit.i ], [ inttoptr (i64 8 to ptr), %start ], [ %raw_ptr.i.i.i.i.i, %"_ZN63_$LT$alloc..alloc..Global$u20$as$u20$core..alloc..Allocator$GT$6shrink17h07a4bd2cb2e9e15aE.exit.i.i.i.i" ] 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          247:  %12 = insertvalue { ptr, i64 } poison, ptr %value.sroa.0.0.copyload.i.i, 0 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          248:  %13 = insertvalue { ptr, i64 } %12, i64 %_2.i.i.i.i.i.i.i.i.i.i.i.i.i.i8.i, 1 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          249:  call void @llvm.lifetime.end.p0(i64 24, ptr nonnull %_2.i) 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          250:  ret { ptr, i64 } %13 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~
          251: } 
check:44'0     ~~
          252:  
check:44'0     ~
          253: declare i32 @__CxxFrameHandler3(...) unnamed_addr #1 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          254:  
check:44'0     ~
          255: ; Function Attrs: mustprogress nocallback nofree nosync nounwind willreturn memory(inaccessiblemem: readwrite) 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          256: declare void @llvm.assume(i1 noundef) #2 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          257:  
check:44'0     ~
          258: ; Function Attrs: nounwind allockind("alloc,uninitialized,aligned") allocsize(0) uwtable 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          259: declare noalias noundef ptr @__rust_alloc(i64 noundef, i64 allocalign noundef) unnamed_addr #3 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          260:  
check:44'0     ~
          261: ; Function Attrs: nounwind allockind("realloc,aligned") allocsize(3) uwtable 
check:44'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            .
            .
>>>>>>
------------------------------------------
---
                                         
Total duration:                        0s
-----------------------------------------
Traceback (most recent call last):
  File "C:\a\rust\rust\src\ci\stage-build.py", line 971, in run
    raise e
  File "C:\a\rust\rust\src\ci\stage-build.py", line 968, in run
    execute_build_pipeline(timer, pipeline, runner, build_args)
  File "C:\a\rust\rust\src\ci\stage-build.py", line 864, in execute_build_pipeline
    pipeline.run_tests()
  File "C:\a\rust\rust\src\ci\stage-build.py", line 186, in run_tests
    cmd(args=[
  File "C:\a\rust\rust\src\ci\stage-build.py", line 543, in cmd
    return subprocess.run(args, env=environment, check=True)
  File "C:\hostedtoolcache\windows\Python\3.11.3\x64\Lib\subprocess.py", line 571, in run
  File "C:\hostedtoolcache\windows\Python\3.11.3\x64\Lib\subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['C:\\hostedtoolcache\\windows\\Python\\3.11.3\\x64\\python.exe', 'C:\\a\\rust\\rust\\x.py', 'test', '--stage', '0', 'tests/assembly', 'tests/codegen', 'tests/codegen-units', 'tests/incremental', 'tests/mir-opt', 'tests/pretty', 'tests/run-pass-valgrind', 'tests/ui', '--exclude', 'tests/ui/process/nofile-limit.rs']' returned non-zero exit status 1.

During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "C:\a\rust\rust\src\ci\stage-build.py", line 981, in <module>
  File "C:\a\rust\rust\src\ci\stage-build.py", line 981, in <module>
    run(runner)
  File "C:\a\rust\rust\src\ci\stage-build.py", line 974, in run
    print_free_disk_space(pipeline)
  File "C:\a\rust\rust\src\ci\stage-build.py", line 815, in print_free_disk_space
    usage = shutil.disk_usage(pipeline.opt_artifacts())
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\hostedtoolcache\windows\Python\3.11.3\x64\Lib\shutil.py", line 1350, in disk_usage
    total, free = nt._getdiskusage(path)
                  ^^^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [WinError 3] The system cannot find the path specified
```

</details>

--- a/src/bootstrap/lib.rs
+++ b/src/bootstrap/lib.rs
@@ -222,6 +222,7 @@ pub struct Build {
     initial_cargo: PathBuf,
     initial_lld: PathBuf,
     initial_libdir: PathBuf,
+    initial_sysroot: PathBuf,
 
     // Runtime state filled in later on
     // C/C++ compilers and archiver for all targets
@@ -389,13 +390,16 @@ impl Build {
             "/dummy".to_string()
         } else {
             output(Command::new(&config.initial_rustc).arg("--print").arg("sysroot"))
-        };
+        }
+        .trim()
+        .to_string();
+
         let initial_libdir = initial_target_dir
             .parent()
             .unwrap()
             .parent()
             .unwrap()
-            .strip_prefix(initial_sysroot.trim())
+            .strip_prefix(&initial_sysroot)
             .unwrap()
             .to_path_buf();
 
@@ -425,6 +429,7 @@ impl Build {
             initial_cargo: config.initial_cargo.clone(),
             initial_lld,
             initial_libdir,
+            initial_sysroot: initial_sysroot.into(),
             local_rebuild: config.local_rebuild,
             fail_fast: config.cmd.fail_fast(),
             doc_tests: config.cmd.doc_tests(),

--- a/src/bootstrap/test.rs
+++ b/src/bootstrap/test.rs
@@ -1424,7 +1424,15 @@ note: if you're sure you want to do this, please open an issue as to why. In the
 
         cmd.arg("--src-base").arg(builder.src.join("tests").join(suite));
         cmd.arg("--build-base").arg(testdir(builder, compiler.host).join(suite));
-        cmd.arg("--sysroot-base").arg(builder.sysroot(compiler));
+
+        // When top stage is 0, that means that we're testing an externally provided compiler.
+        // In that case we need to use its specific sysroot for tests to pass.
+        let sysroot = if builder.top_stage == 0 {
+            builder.initial_sysroot.clone()
+        } else {
+            builder.sysroot(compiler).to_path_buf()
+        };
+        cmd.arg("--sysroot-base").arg(sysroot);
         cmd.arg("--stage-id").arg(stage_id);
         cmd.arg("--suite").arg(suite);
         cmd.arg("--mode").arg(mode);

--- a/src/ci/stage-build.py
+++ b/src/ci/stage-build.py
@@ -124,6 +124,12 @@ class Pipeline:
     def metrics_path(self) -> Path:
         return self.build_root() / "build" / "metrics.json"
 
+    def executable_extension(self) -> str:
+        raise NotImplementedError
+
+    def skipped_tests(self) -> Iterable[str]:
+        return ()
+
 
 class LinuxPipeline(Pipeline):
     def checkout_path(self) -> Path:
@@ -151,6 +157,13 @@ class LinuxPipeline(Pipeline):
 
     def supports_bolt(self) -> bool:
         return True
+
+    def executable_extension(self) -> str:
+        return ""
+
+    def skipped_tests(self) -> Iterable[str]:
+        # This test fails because of linker errors, as of June 2023.
+        yield "tests/ui/process/nofile-limit.rs"
 
 
 class WindowsPipeline(Pipeline):
@@ -210,6 +223,13 @@ class WindowsPipeline(Pipeline):
 
     def supports_bolt(self) -> bool:
         return False
+
+    def executable_extension(self) -> str:
+        return ".exe"
+
+    def skipped_tests(self) -> Iterable[str]:
+        # This test fails as of June 2023
+        yield "tests\\codegen\\vec-shrink-panik.rs"
 
 
 def get_timestamp() -> float:
@@ -403,9 +423,9 @@ def delete_directory(path: Path):
     shutil.rmtree(path)
 
 
-def unpack_archive(archive: Path):
+def unpack_archive(archive: Path, target_dir: Optional[Path] = None):
     LOGGER.info(f"Unpacking archive `{archive}`")
-    shutil.unpack_archive(archive)
+    shutil.unpack_archive(str(archive), extract_dir=str(target_dir) if target_dir is not None else None)
 
 
 def download_file(src: str, target: Path):
@@ -455,6 +475,7 @@ def cmd(
             )
     return subprocess.run(args, env=environment, check=True)
 
+
 class BenchmarkRunner:
     def run_rustc(self, pipeline: Pipeline):
         raise NotImplementedError
@@ -464,6 +485,7 @@ class BenchmarkRunner:
 
     def run_bolt(self, pipeline: Pipeline):
         raise NotImplementedError
+
 
 class DefaultBenchmarkRunner(BenchmarkRunner):
     def run_rustc(self, pipeline: Pipeline):
@@ -478,6 +500,7 @@ class DefaultBenchmarkRunner(BenchmarkRunner):
                 LLVM_PROFILE_FILE=str(pipeline.rustc_profile_template_path())
             )
         )
+
     def run_llvm(self, pipeline: Pipeline):
         run_compiler_benchmarks(
             pipeline,
@@ -493,6 +516,7 @@ class DefaultBenchmarkRunner(BenchmarkRunner):
             scenarios=["Full"],
             crates=LLVM_BOLT_CRATES
         )
+
 
 def run_compiler_benchmarks(
         pipeline: Pipeline,
@@ -650,9 +674,7 @@ def gather_llvm_profiles(pipeline: Pipeline, runner: BenchmarkRunner):
 def gather_rustc_profiles(pipeline: Pipeline, runner: BenchmarkRunner):
     LOGGER.info("Running benchmarks with PGO instrumented rustc")
 
-
     runner.run_rustc(pipeline)
-
 
     profile_path = pipeline.rustc_profile_merged_file()
     LOGGER.info(f"Merging Rustc PGO profiles to {profile_path}")
@@ -770,6 +792,86 @@ def record_metrics(pipeline: Pipeline, timer: Timer):
     log_metrics(metrics)
 
 
+def run_tests(pipeline: Pipeline):
+    """
+    After `dist` is executed, we extract its archived components into a sysroot directory,
+    and then use that extracted rustc as a stage0 compiler.
+    Then we run a subset of tests using that compiler, to have a basic smoke test which checks
+    whether the optimization pipeline hasn't broken something.
+    """
+    build_dir = pipeline.build_root() / "build"
+    dist_dir = build_dir / "dist"
+
+    def extract_dist_dir(name: str) -> Path:
+        target_dir = build_dir / "optimized-dist"
+        target_dir.mkdir(parents=True, exist_ok=True)
+        unpack_archive(dist_dir / f"{name}.tar.xz", target_dir=target_dir)
+        extracted_path = target_dir / name
+        assert extracted_path.is_dir()
+        return extracted_path
+
+    # Extract rustc, libstd, cargo and src archives to create the optimized sysroot
+    rustc_dir = extract_dist_dir(f"rustc-nightly-{PGO_HOST}") / "rustc"
+    libstd_dir = extract_dist_dir(f"rust-std-nightly-{PGO_HOST}") / f"rust-std-{PGO_HOST}"
+    cargo_dir = extract_dist_dir(f"cargo-nightly-{PGO_HOST}") / f"cargo"
+    extracted_src_dir = extract_dist_dir("rust-src-nightly") / "rust-src"
+
+    # We need to manually copy libstd to the extracted rustc sysroot
+    shutil.copytree(
+        libstd_dir / "lib" / "rustlib" / PGO_HOST / "lib",
+        rustc_dir / "lib" / "rustlib" / PGO_HOST / "lib"
+    )
+
+    # Extract sources - they aren't in the `rustc-nightly-{host}` tarball, so we need to manually copy libstd
+    # sources to the extracted sysroot. We need sources available so that `-Zsimulate-remapped-rust-src-base`
+    # works correctly.
+    shutil.copytree(
+        extracted_src_dir / "lib" / "rustlib" / "src",
+        rustc_dir / "lib" / "rustlib" / "src"
+    )
+
+    rustc_path = rustc_dir / "bin" / f"rustc{pipeline.executable_extension()}"
+    assert rustc_path.is_file()
+    cargo_path = cargo_dir / "bin" / f"cargo{pipeline.executable_extension()}"
+    assert cargo_path.is_file()
+
+    config_content = f"""profile = "user"
+changelog-seen = 2
+
+[build]
+rustc = "{rustc_path.as_posix()}"
+cargo = "{cargo_path.as_posix()}"
+
+[llvm]
+download-ci-llvm = true
+"""
+    logging.info(f"Using following `config.toml` for running tests:\n{config_content}")
+
+    # Simulate a stage 0 compiler with the extracted optimized dist artifacts.
+    with open("config.toml", "w") as f:
+        f.write(config_content)
+
+    args = [
+        sys.executable,
+        pipeline.checkout_path() / "x.py",
+        "test",
+        "--stage", "0",
+        "tests/assembly",
+        "tests/codegen",
+        "tests/codegen-units",
+        "tests/incremental",
+        "tests/mir-opt",
+        "tests/pretty",
+        "tests/run-pass-valgrind",
+        "tests/ui",
+        ]
+    for test_path in pipeline.skipped_tests():
+        args.extend(["--exclude", test_path])
+    cmd(args=args, env=dict(
+        COMPILETEST_FORCE_STAGE0="1"
+    ))
+
+
 def execute_build_pipeline(timer: Timer, pipeline: Pipeline, runner: BenchmarkRunner, final_build_args: List[str]):
     # Clear and prepare tmp directory
     shutil.rmtree(pipeline.opt_artifacts(), ignore_errors=True)
@@ -843,6 +945,11 @@ def execute_build_pipeline(timer: Timer, pipeline: Pipeline, runner: BenchmarkRu
     with timer.section("Stage 4 (final build)") as stage4:
         cmd(final_build_args)
         record_metrics(pipeline, stage4)
+
+    # Try builds can be in various broken states, so we don't want to gatekeep them with tests
+    if not is_try_build():
+        with timer.section("Run tests"):
+            run_tests(pipeline)
 
 
 def run(runner: BenchmarkRunner):

